src/override/rust.cc

```diff
@@ -9,47 +9,199 @@
 
 using namespace snmalloc;
 
-extern "C" SNMALLOC_EXPORT void*
-  SNMALLOC_NAME_MANGLE(rust_alloc)(size_t alignment, size_t size)
+namespace
 {
-  return ThreadAlloc::get().alloc(aligned_size(alignment, size));
-}
+  enum class SizeChange
+  {
+    Unknown,
+    Grow,
+    Shrink
+  };
+  template<ZeroMem zero_mem, SizeChange size_change>
+  SNMALLOC_FAST_PATH_INLINE void* realloc_helper(
+    Alloc& alloc,
+    void* ptr,
+    size_t old_alignment,
+    size_t old_size,
+    size_t new_alignment,
+    size_t new_size)
+  {
+    size_t aligned_old_size = aligned_size(old_alignment, old_size),
+           aligned_new_size = aligned_size(new_alignment, new_size);
+    if (
+      size_to_sizeclass_full(aligned_old_size).raw() ==
+      size_to_sizeclass_full(aligned_new_size).raw())
+      return ptr;
```

Comment:
```suggestion
    {
      return ptr;
    }
```

As a general style point we should have braces around all `if` statement bodies because it reduces bugs introduced when someone edits the code later, but this case is a particularly good example of needing them for readability because the wrapped line in the condition is at the same indent level as the body of the conditiona.